stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1082,6 +1082,43 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         
     }
     
+    // slightly faster paths for common sequences
+    
+    public init<S: Sequence>(_ elements: S) where S.Iterator.Element == UInt8 {
+        let underestimatedCount = elements.underestimatedCount
+        self.init(count: underestimatedCount)
+        var idx = 0
+        for byte in elements {
```

Comment:
If we're already add it, we can might be able to speed this up a bit more by factoring out the conditional from this loop (and write two loops). Write the first `underestimatedCount` bytes directly, then loop over the remaining and append. Don't know if the optimizer can optimize that even more, but just a thought.